### ISSUE TITLE
models/research/skip_thoughts/skip_thoughts/skip_thoughts_model.py:Why does the skip thought vector was put in the initial_state in the function build_decoders

### ISSUE THREAD

**Issue by IronE-G-G** [tensorflow/models#5909]
Hello,
I wonder whether there is a mistake in the decoder of skipthoughts. 
The file is **tensorflow/models/research/skip_thoughts/skip_thoughts/skip_thoughts_model.py**

### Describe the problem
Why does the skip thought vector was put in the initial_state in the function **build_decoders** when it use the function **_build_decoder** in **line 333**? I have read the skip-thought paper and the skip-thought vector was input in every time step in the decoder in the paper.  
### code
<code>
  def _build_decoder(self, name, embeddings, targets, mask, initial_state,
                     reuse_logits):
...
</code>

    
<code>
  def build_decoders(self):
  
    """Builds the sentence decoders.

    Inputs:
      self.decode_pre_emb
      self.decode_post_emb
      self.decode_pre_ids
      self.decode_post_ids
      self.decode_pre_mask
      self.decode_post_mask
      self.thought_vectors

    Outputs:
      self.target_cross_entropy_losses
      self.target_cross_entropy_loss_weights
    """
    if self.mode != "encode":
      # Pre-sentence decoder.
      self._build_decoder("decoder_pre", self.decode_pre_emb,
                          self.decode_pre_ids, self.decode_pre_mask,
                          self.thought_vectors, False)

      # Post-sentence decoder. Logits weights are reused.
      self._build_decoder("decoder_post", self.decode_post_emb,
                          self.decode_post_ids, self.decode_post_mask,
                          self.thought_vectors, True)
</code>

### system information
**What is the top-level directory of the model you are using tensorflow/models/research/skip_thoughts**
**Have I written custom code:No**
**OS Platform and Distribution: N/A** 
**TensorFlow installed from: N/A**
**TensorFlow version :N/A**
**Bazel version :N/A**
**CUDA/cuDNN version: NO**
**GPU model and memory: NO**

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
